prov/mrail/src/mrail_init.c

```diff
@@ -134,20 +285,16 @@ int mrail_get_core_info(uint32_t version, const char *node, const char *service,
 		       "--- End fi_getinfo for rail: %zd ---\n", i);
 		if (ret)
 			goto err;
-
```

Comment:
@wckzhang  I think building the mrail_info_vec here would be cleaner. Something like the following pseudo-code:
```
for (i = 0; mrail_addr_strv[i]; i++):
        ...
	fi_getinfo(..., &rail_info_list)	// already there in current code
	if (!i):
		mrail_num_info = num_info(rail_info_list)
        else:
                assert(mrail_num_info == num_info(rail_info_list))
	for i in mrail_num_info:
		assert(mrail_info_vec[i].fabric_attr->prov_name == rail_info_list->fabric_attr->prov_name)
		mrail_info_vec[i].append(rail_info_list.pop_front())
```
Thoughts?

Then prefix the mrail header infos.